pandas/tests/test_nanops.py

```diff
@@ -182,12 +182,17 @@ def _coerce_tds(targ, res):
                                    check_dtype=check_dtype)
 
     def check_fun_data(self, testfunc, targfunc, testarval, targarval,
-                       targarnanval, check_dtype=True, **kwargs):
+                       targarnanval, check_dtype=True, empty_targfunc=None,
+                       **kwargs):
         for axis in list(range(targarval.ndim)) + [None]:
             for skipna in [False, True]:
                 targartempval = targarval if skipna else targarnanval
-                try:
+                if skipna and empty_targfunc and pd.isna(targartempval).all():
```

Comment:
I'm reasonably sure I refactored these correctly, but would appreciate a close review here. `empty_targfunc` should be like `nansum` or `nanprod`, in which case we want to call it on `targartempval`. I should double check why just passing `nansum` for `targfunc` didn't work though...

Reply:
use the imported ``isna``